src/effective-dart/usage.md

```diff
@@ -350,13 +350,19 @@ then it probably does make sense to have a separate boolean field.
 
 Checking that a nullable variable is not equal to `null` promotes the variable
 to a non-nullable type. That lets you access members on the variable and pass it
-to functions expecting a non-nullable type. Unfortunately, promotion is only
-sound for local variables and parameters, so fields and top-level variables
-aren't promoted.
+to functions expecting a non-nullable type.
 
-One pattern to work around this is to assign the field's value to a local
-variable. Null checks on that variable do promote, so you can safely treat
-it as non-nullable.
+Type promotion is only supported, however, for local variables, parameters, and
+private final fields. Values that are open to manipulation, whether
+directly, like top-level variables, or indirectly, like fields with a concrete
+getter of the same name in the same library, cannot be type promoted.
```

Comment:
> "indirectly, like fields with a concrete getter of the same name in the same library"

While Effective Dart is best read after finishing the language tour, it is still targeted at those newer to the language and/or programming. I think this sentence/concept might be too abstract or confusing for some of its targeted readers.

With extra context on why and an inline code example, such as a link to the corresponding entry in your updated non-promotion doc, it might be fine though.

However, since readers often won't make that page switch and it doesn't seem super essential for the point of this Effective Dart entry, what do you think about dropping the expansion of "Values that are open to manipulation"?

If you have related docs in "Understanding null safety" or type system docs (now or in future), perhaps a cross link to there could be added as an alternative.

Reply:
> what do you think about dropping the expansion of "Values that are open to manipulation"?

Done 👍 

> If you have related docs in "Understanding null safety" or type system docs (now or in future), perhaps a cross link to there could be added as an alternative.

I think the "fixing type promotion page" is the most complete list of reasons you can't type promote, so I treated your other comment about linking to that page in combination with this one: I removed the explicit "check out <that page>..." link and instead linked to it over the text `Values that are open to manipulation [can't be type promoted][].`